src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java

```diff
@@ -144,4 +153,58 @@ public FileDownloadController(ExperimentFileLocationService experimentFileLocati
             zipOutputStream.close();
         }
     }
+
+    @RequestMapping(value = "json/experiments/check/zip",
+            method = RequestMethod.GET,
+            produces = "application/json;charset=UTF-8")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public String
+    checkMultipleExperimentsFileValid(@RequestParam(value = "accession", defaultValue = "") List<String> accessions)
+    {
+        List<Experiment> experiments = new ArrayList<>();
+        for (var accession : accessions) {
+            try {
+                experiments.add(experimentTrader.getPublicExperiment(accession));
+            } catch (Exception e) {
+                LOGGER.debug("Invalid experiment accession: {}", accession);
+            }
+        }
+        List<ExperimentFileType> fileTypeCheckList = new ArrayList<>(Arrays.asList(
+                ExperimentFileType.QUANTIFICATION_RAW,
+                ExperimentFileType.NORMALISED,
+                ExperimentFileType.EXPERIMENT_DESIGN));
+
+       Map<String, List<String>> inValidFilesList = new HashMap<>();
+
+        if (!experiments.isEmpty()) {
```

Comment:
Why are you checking this? What will happen if you remove this check?

Reply:
This is the same logic as line 124. In case invalid experiment accession resulting in empty experiment list. It will trigger errors in the following codes.

Reply:
If there were invalid experiment accessions they were skipped when you built `experiments`, so you shouldn’t be concerned about it, right?

Reply:
This check is in case no valid experiment is in the list at all and then we skip the following logic.